src/sysfiles.c

```diff
@@ -3589,6 +3589,67 @@ Obj SyReadStringFid(Int fid) {
 #endif
 
 
+#ifdef USE_CUSTOM_MEMMOVE
+// The memmove in glibc on 32-bit SSE2 systems, contained in
+// __memmove_sse2_unaligned, is buggy in at least versions
+// 2.21 - 2.26 when crossing the 2GB boundary, so GAP must
+// include its own simple memmove implementation.
+void * SyMemmove(void * dst, const void * src, UInt size)
+{
+    char *       d = dst;
+    const char * s = src;
+
+    if ((d == s) || (size == 0))
+        return dst;
+
+    if (d + size < s || d > s + size) {
+        memcpy(dst, src, size);
+    }
+    else if (d > s) {
+        d = d + (size - 1);
+        s = s + (size - 1);
+
+        // This is about 4x slower than glibc, but
+        // is simple and also complicated enough that
+        // gcc or clang seem unable to "optimise" it back
+        // into a call to memmove.
```

Comment:
Under which condition did you test this? E.g. did you also try it with `-O3`?

Reply:
I tried with `-O3` and `-flto`. I checked using `objdump -t` that the resulting GAP executable doesn't have any references to `memmove`.